pkg/rpc/adminservice/base.go

```diff
@@ -135,8 +138,17 @@ func NewAdminServer(kubeConfig, master string) *AdminService {
 	scheduledWorkflowExecutor := workflowScheduler.GetWorkflowExecutor(executionManager, launchPlanManager)
 	logger.Info(context.Background(), "Successfully initialized a new scheduled workflow executor")
 	go func() {
+		logger.Info(context.Background(), "Starting the scheduled workflow executor")
 		scheduledWorkflowExecutor.Run()
-		logger.Info(context.Background(), "Successfully started running the scheduled workflow executor")
+
+		maxReconnectAttempts := configuration.ApplicationConfiguration().GetSchedulerConfig().
+			WorkflowExecutorConfig.ReconnectAttempts
+		for reconnectAttempt := 0; reconnectAttempt < maxReconnectAttempts; reconnectAttempt++ {
+			time.Sleep(workflowExecutorReconnectDelay)
```

Comment:
Why is the delay? do we need to add Jitter?

Reply:
yes, but it's just an (unproven) idea. i have no idea what's causing the client connection to terminate so i figured it might be useful. if you disagree i can remove altogether

Reply:
does it connect to the pubsub, on AWS SQS?

Reply:
yeah